MaterialLib/MPL/Properties/PermeabilityMohrCoulombFailureIndexModel.cpp

```diff
@@ -0,0 +1,148 @@
+            "given in the 'Solid' phase, not in '{:s}' phase.",
+            phase->name);
+    }
+}
+
+template <int DisplacementDim>
+PropertyDataType
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& pos, double const t,
+    double const /*dt*/) const
+{
+    constexpr int symmetric_tensor_size =
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    using SymmetricTensor = Eigen::Matrix<double, symmetric_tensor_size, 1>;
+
+    auto const& stress_vector = std::get<SymmetricTensor>(
+        variable_array[static_cast<int>(Variable::stress)]);
+    auto const& stress_tensor =
+        MathLib::KelvinVector::symmetricTensorToFullTensor(stress_vector);
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>>
+        eigenvalue_solver(stress_tensor);
+
+    // Principle stress
+    auto const sigma = eigenvalue_solver.eigenvalues();
+
+    double const f =
+        std::fabs(0.5 * (sigma[0] - sigma[2]) /
+                  (c_ + 0.5 * (sigma[0] + sigma[2]) * std::tan(phi_)));
+
+    auto k_data = k0_(t, pos);
+
+    if (f > 1.0)
+    {
+        for (auto& k_i : k_data)
+        {
+            k_i += kr_ * std::exp(b_ * f);
+        }
+    }
+
+    // Local coordinate  transformation is only applied for the initial
+    // intrinsic permeability with orthotropic assumption.
+    if (local_coordinate_system_ &&
+        ((k_data.size() == 2) || (k_data.size() == 3)))
+    {
+        assert(k_data.size() == DisplacementDim);
+
+        Eigen::Matrix<double, DisplacementDim, DisplacementDim> const e =
+            local_coordinate_system_->transformation<DisplacementDim>(pos);
+        Eigen::Matrix<double, DisplacementDim, DisplacementDim> k =
+            Eigen::Matrix<double, DisplacementDim, DisplacementDim>::Zero();
+
+        for (int i = 0; i < DisplacementDim; ++i)
+        {
+            Eigen::Matrix<double, DisplacementDim, DisplacementDim> const
+                ei_otimes_ei = e.col(i) * e.col(i).transpose();
+
+            k += k_data[i] * ei_otimes_ei;
+        }
+        return k;
+    }
+
+    auto const& k0_variant = fromVector(k_data);
+
+    PropertyDataType k = std::visit(
+        [&](auto const& k0_variant) -> PropertyDataType {
+            using T = std::decay_t<decltype(k0_variant)>;
+            if
+                constexpr(std::is_same_v<T, double>) { return k0_variant; }
+            else
+            {
+                return k0_variant.eval();
+            }
+        },
+        k0_variant);
+
+    return k;
+}
+
+template <int DisplacementDim>
+PropertyDataType
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::dValue(
+    VariableArray const& /*variable_array*/, Variable const /*variable*/,
+    ParameterLib::SpatialPosition const& /*pos*/, double const /*t*/,
+    double const /*dt*/) const
+{
+    // no need to compute the derivative.
+    return 0.0;
```

Comment:
OGS_FATAL("Derivative not implemented for etc. and so forth...");
```suggestion
    OGS_FATAL("Derivative not implemented for etc. and so forth...");
```

Reply:
Changed with a more detailed info.